src/wrappers.js

```
@@ -236,3 +236,214 @@ wrappers[".google.protobuf.ListValue"] = {
         return this.toObject(message, options);
     }
 };
+
+// Custom wrapper for Duration
+wrappers[".google.protobuf.Duration"] = {
+    fromObject: function(object) {
+        // If already a Duration instance, return as is
+        if (object instanceof this.ctor) return object;
+        
+        // Handle string input (e.g., "1.5s", "2m", "1h", "1h30m", "2d5h30m15s")
+        if (typeof object === "string") {
+            // Parse compound duration string like "1h30m15s" or "2d5h30m15s"
+            // Note: Multiple segments of the same unit are allowed and will be added together
+            // e.g., "2s32.232s" becomes "34.232s"
+            var totalSeconds = 0;
+            var totalNanos = 0;
+            var sign = 1;
+            
+            // Check for negative duration
+            if (object.startsWith('-')) {
+                sign = -1;
+                object = object.substring(1);
+            }
+            
+            // Match all duration parts (e.g., "1h", "30m", "15s")
+            var parts = object.match(/(\d+(?:\.\d+)?)([smhd])/g);
+            if (!parts || parts.length === 0) {
+                throw new Error("Invalid duration format. Expected format: '1.5s', '2m', '1h', '1h30m', '2d5h30m15s', etc.");
+            }
+                        
+            // Track units used for validation/warning
+            var unitsUsed = { s: 0, m: 0, h: 0, d: 0 };
+            
+            for (var i = 0; i < parts.length; i++) {
+                var part = parts[i];
+                var match = part.match(/(\d+(?:\.\d+)?)([smhd])/);
+                if (!match) continue;
```

Since we are iterating on parts one at a time, for cases where the string was only partially valid (for eg: `1h43GHI`), we'll end respecting valid parts and silently ignore the invalid ones. I think we should throw an error if the entire string is not valid.